docs/concepts/alias.md

```diff
@@ -3,9 +3,164 @@ An alias is an alternative name for a field, used when serializing and deseriali
+    [`pydantic.aliases.AliasChoices`][pydantic.aliases.AliasChoices]<br>
+
+Pydantic provides two special types for convenience when using `validation_alias`: `AliasPath` and `AliasChoices`.
+
+The `AliasPath` is used to specify a path to a field using aliases. For example:
+
+```py lint="skip"
+from pydantic import BaseModel, Field, AliasPath
+
+
+class User(BaseModel):
+    first_name: str = Field(validation_alias=AliasPath('names', 0))
+    last_name: str = Field(validation_alias=AliasPath('names', 1))
+
+user = User.model_validate({'names': ['John', 'Doe']})  # (1)!
+print(user)
+#> first_name='John' last_name='Doe'
+```
+
+1. We are using `model_validate` to validate a dictionary using the field aliases.
+
+    You can see more details about [`model_validate`][pydantic.main.BaseModel.model_validate] in the API reference.
+
+In the `'first_name'` field, we are using the alias `'names'` and the index `0` to specify the path to the first name.
+In the `'last_name'` field, we are using the alias `'names'` and the index `1` to specify the path to the last name.
+
+`AliasChoices` is used to specify a choice of aliases. For example:
+
+```py lint="skip"
+from pydantic import BaseModel, Field, AliasChoices
+
+
+class User(BaseModel):
+    first_name: str = Field(validation_alias=AliasChoices('first_name', 'fname'))
+    last_name: str = Field(validation_alias=AliasChoices('last_name', 'lname'))
+
+user = User.model_validate({'fname': 'John', 'lname': 'Doe'})  # (1)!
+print(user)
+#> first_name='John' last_name='Doe'
+user = User.model_validate({'first_name': 'John', 'lname': 'Doe'})  # (2)!
+print(user)
+#> first_name='John' last_name='Doe'
+```
+
+1. We are using the second alias choice for both fields.
+2. We are using the first alias choice for the field `'first_name'` and the second alias choice
+   for the field `'last_name'`.
+
+You can also use `AliasChoices` with `AliasPath`:
+
+```py lint="skip"
+from pydantic import BaseModel, Field, AliasPath, AliasChoices
+
+
+class User(BaseModel):
+    first_name: str = Field(validation_alias=AliasChoices('first_name', AliasPath('names', 0)))
+    last_name: str = Field(validation_alias=AliasChoices('last_name', AliasPath('names', 1)))
+
+
+user = User.model_validate({'first_name': 'John', 'last_name': 'Doe'})
+print(user)
+#> first_name='John' last_name='Doe'
+user = User.model_validate({'names': ['John', 'Doe']})
+print(user)
+#> first_name='John' last_name='Doe'
+user = User.model_validate({'names': ['John'], 'last_name': 'Doe'})
+print(user)
+#> first_name='John' last_name='Doe'
+```
+
+## Using alias generators
+
+You can use the `alias_generator` parameter of [`Config`][pydantic.config.ConfigDict.alias_generator] to specify
+a callable (or group of callables, via `AliasGenerator`) that will generate aliases for all fields in a model.
+This is useful if you want to use a consistent naming convention for all fields in a model, but doin't
```

Comment:
`don't` is misspelled

Reply:
@shaunpatterson feel free to create a PR with a fix, thanks!